### ISSUE TITLE
"in-dim" question

### ISSUE THREAD

**Issue by Michael-l123** [CYBruce/STAWnet#7]
Hello, the "in-dim" in the training code is set to 3, and the dimension of the data generated by "generate_training_data.py" is 1. Is there any other processing needed?

**Comment by manchr180** [comment]
Not only that but also if I set in-dim to 1 I get "Expected 2D (unbatched) or 3D (batched) input to conv1d, but got input of size: [64, 32, 207, 13]" when I run train.py. @Michael-l123 have you found any solution for that?

**Comment by CYBruce** [comment]
Sorry for the delayed reply. Please set the **_add_time_in_day_** and **_add_day_in_week_** to True if you want the in-dim to be 3.

**Comment by debimishra89** [comment]
> Not only that but also if I set in-dim to 1 I get "Expected 2D (unbatched) or 3D (batched) input to conv1d, but got input of size: [64, 32, 207, 13]" when I run train.py. @Michael-l123 have you found any solution for that?

I am getting the same error. Did you find any solution?

**Comment by yjh-bill** [comment]
> > Not only that but also if I set in-dim to 1 I get "Expected 2D (unbatched) or 3D (batched) input to conv1d, but got input of size: [64, 32, 207, 13]" when I run train.py. [@Michael-l123](https://github.com/Michael-l123) have you found any solution for that?
> 
> I am getting the same error. Did you find any solution?

It's a pytorch version issue. 

See [https://github.com/YuvalNirkin/fsgan/issues/162](https://github.com/YuvalNirkin/fsgan/issues/162) or [https://github.com/nnzhan/Graph-WaveNet/issues/34](https://github.com/nnzhan/Graph-WaveNet/issues/34)
